webapp-django/crashstats/crashstats/views.py

```diff
@@ -2734,8 +2731,21 @@ def correlations_json(request):
     signature = form.cleaned_data['signature']
 
     api = models.Correlations()
-    return api.get(report_type=report_type, product=product, version=version,
-                   platform=platform, signature=signature)
+    context = {}
+    for report_type in report_types:
+        # To keep things simple,
+        # the actual middleware query only supports querying by 1 report
+        # type at a time. At some point we ought to change that so it
+        # takes in a list instead.
+        # One big change at a time!
+        context[report_type] = api.get(
+            report_type=report_type,
+            product=product,
+            version=version,
+            platform=platform,
+            signature=signature
+        )
```

Comment:
Places like this I want to change too. Maybe. That'd reduced the number of queries to the middleware. 

However, making many small queries is sometimes better than one big monster query. After all, the way correlations work (under the Correlations tab) is that it loads the 3 most interesting kinds and then you can manually extra load other ones one at a time.